azurerm/internal/services/compute/shared_image_resource.go

```diff
@@ -344,3 +374,53 @@ func flattenGalleryImageIdentifier(input *compute.GalleryImageIdentifier) []inte
 		},
 	}
 }
+
+func expandGalleryImagePurchasePlan(input []interface{}) *compute.ImagePurchasePlan {
+	if len(input) == 0 {
```

Comment:
We'll want to check that input[0] doesn't equal nil as well so we don't panic
```suggestion
	if len(input) == 0 && input[0] != nil {
```

Reply:
I suppose this should be
```suggestion
	if len(input) == 0 && input[0] == nil {
```
if we are doing a nil check here?

Reply:
And I suppose since the `name` attribute in this block is required, the situation `input[0] == nil` will never happen, right?

Reply:
But I still added this check for safety

Reply:
Oh yes! Thanks for catching that `||` instead of `&&`. We should check even if should never happen as a just in case.